x-pack/test/ui_capabilities/common/services/ui_capabilities.ts

```diff
@@ -41,22 +40,23 @@ export class UICapabilitiesService {
     });
   }
 
-  public async get(
-    credentials: BasicCredentials | null,
-    spaceId?: string
-  ): Promise<GetUICapabilitiesResult> {
+  public async get({
+    credentials,
+    navLinks,
+    spaceId,
+  }: {
+    credentials?: BasicCredentials;
+    navLinks?: Record<string, boolean>;
```

Comment:
This works reasonably well for the time being, as more capabilities are moved to being determined based on the application service, it feels like this approach is going to need to be adapted. Do you have any ideas in mind, or should we just address this when we get to that point?

The primary motivation of these tests was to ensure that spaces and security could work collaboratively to turn enable/disable certain ui capabilities, perhaps we should just be performing this same logic using completely mocked plugins and a static list of capabilities?

Otherwise, there's potential that we can use the list of "features" to drive this functionality as well in the future...

Reply:
Since most of the logic isn't tightly coupled to the specific uiCapabilities, I think a mocked list or would work fine. Alternatively, we could define uiCapabilities as part of a plugin's `kibana.json` file in the new platform rather than being tied to applications. Downside to this is there are some capabilities that are dynamically derived from other data (`savedObjectManagement`, for example).

I'd actually considered changing this from `navLinks` to `applications` to make it more clear that turning those off disables the entire application, not just the link in the navigation. I've opted to put that into a separate change since this PR keeps ballooning already.